crates/rattler_installs_packages/src/index/package_database.rs

```diff
@@ -472,6 +485,54 @@ impl PackageDb {
         Ok(None)
     }
 
+    async fn get_metadata_stree<'a, A: Borrow<ArtifactInfo>>(
+        &self,
+        artifacts: &'a [A],
+        wheel_builder: &WheelBuilder,
+    ) -> miette::Result<Option<(&'a A, WheelCoreMetadata)>> {
+        let stree = artifacts
+            .iter()
+            .filter(|artifact_info| (*artifact_info).borrow().is::<STree>());
+
+        // Keep track of errors
+        // only print these if we have not been able to find any metadata
+        let mut errors = Vec::new();
+        for ai in stree {
+            let artifact_info: &ArtifactInfo = ai.borrow();
+            let stree_name = artifact_info.filename.as_stree().unwrap_or_else(|| {
```

Comment:
You can use the newly add `is::<STree>` here :) Its a bit shorter

Reply:
I can use `as_inner` but it's seems the same to me ( here I need to extract the distribution name itself )